src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -380,8 +424,10 @@ public static OptionData fromData(@Nonnull DataObject json)
                 choices1.stream(DataArray::getObject).forEach(o ->
                 {
                     Object value = o.get("value");
-                    if (value instanceof Number)
+                    if (value instanceof Long)
                         option.addChoice(o.getString("name"), ((Number) value).intValue());
+                    else if (value instanceof Double)
```

Comment:
Can this also be instanceof Float? (see etf, jackson inconsistency, etc.)

Reply:
Maybe we should do:

```java
if (value instanceof Float || value instanceof Double)
  ...
else if (value instanceof Number) // can only be int type
  ...
```

or

```java
if (o.isType("value", FLOAT))
  ...
else if (o.isType("value", INTEGER))
  ...
```

Make sure to actually test that these checks work correctly

Reply:
Not sure how I would go about testing those...
Btw, the tests will probably also require some tinkering as they only test for strings and longs right now but not floats/doubles.